src/Language/Fixpoint/Solver/Solution.hs

```diff
@@ -292,7 +302,10 @@ type ExprInfo    = (F.Expr, KInfo)
 apply :: CombinedEnv -> Sol.Sol a Sol.QBind -> F.IBindEnv -> ExprInfo
 apply g s bs      = (F.conj (pks:ps), kI)   -- see [NOTE: pAnd-SLOW]
   where
-    (pks, kI)     = applyKVars g s ks  
+    -- Clear the "known" bindings for applyKVars, since it depends on
+    -- using the fully expanded representation of the predicates to bind their
+    -- variables with quantifiers.
```

Comment:
This is a shortcoming that wouldn't occur should we implement the caching described in approach (1) of #494. That said, the overhead seems pretty small currently to go for it.